### PR TITLE
Split ieeg_labelElectrodesDestrieux.m

### PR DESCRIPTION
**Pull request by hharveygit**
Created ieeg_getLabelXyzDestrieux.m from ieeg_labelElectrodesDestrieux.m
- returns Destrieux labels for input XYZ coordinates

Modified ieeg_labelEletrodesDestrieux.m to call ieeg_getLabelXyzDestrieux.m

^ Both files tested to match outputs from original Destrieux labels.

Also added helper function ieeg_getPairXyzs.m
- returns interpolated XYZ coordinates at the midpoint of electrode pairs

### CODE REVIEW

File: functions/ieeg_labelElectrodesDestrieux.m [hunk 1]
```diff
@@ -1,4 +1,4 @@
-function [t_new] = ieeg_labelElectrodesDestrieux(FSdir,electrodes_tsv_name,saveNew,circleradius)
+function [t_new] = ieeg_labelElectrodesDestrieux(FSdir, electrodes_tsv_name, saveNew, circleradius, hemi)
```
Comment (dorahermes) [inline functions/ieeg_labelElectrodesDestrieux.m]: what happens if there are bilateral electrodes?

**Review by dorahermes — COMMENTED** [review]
Only question I have is how this deals with bilateral implants, would we run 2x, once for left, once for right sided electrodes?

**Comment by hharveygit** [comment]
I haven't thought about any bilateral cases yet. Good point. All of the code I've written so far has been under the assumption of unilaterality. I'll have to modify the code a bit to generalize it to a bilateral situation. The original label Destrieux code actually had left hemisphere hard-coded in it.

Best
-Harvey

Sent from my iPhone

On Nov 13, 2020, at 6:10 PM, dorahermes <notifications@github.com> wrote:

﻿

@dorahermes commented on this pull request.

Only question I have is how this deals with bilateral implants, would we run 2x, once for left, once for right sided electrodes?

________________________________

In functions/ieeg_labelElectrodesDestrieux.m<https://github.com/MultimodalNeuroimagingLab/mnl_ieegBasics/pull/5#discussion_r523294878>:

> @@ -1,4 +1,4 @@
-function [t_new] = ieeg_labelElectrodesDestrieux(FSdir,electrodes_tsv_name,saveNew,circleradius)
+function [t_new] = ieeg_labelElectrodesDestrieux(FSdir, electrodes_tsv_name, saveNew, circleradius, hemi)


what happens if there are bilateral electrodes?

File: functions/ieeg_getLabelXyzDestrieux.m [hunk 2]
```diff
@@ -0,0 +1,201 @@
+%
+%   Returns Destrieux labels to input xyz coordinates based on volume labelling
+%   Requires vistasoft
+%
+%   labs = ieeg_getLabelXyzDestrieux(xyzs, FSdir)
+%   [labs, labs_val] = ieeg_getLabelXyzDestrieux(xyzs, FSdir, hemi, rad)
+%       
+%       xyzs =              nx3 array of n [x, y, z] coordinates
+%       FSdir =             path to FreeSurfer directory for the current subject; e.g. */derivatives/freesurfer/sub-<subject>
+%       hemi (optional) =   string: 'left', 'L', 'l' or 'right', etc. corresponding to hemisphere of subject. 'left' by default
```
Comment (dorahermes) [inline functions/ieeg_getLabelXyzDestrieux.m]: hemi is actually only used here to load the Destrieux colortable, which is hemisphere agnostic, but not for any further processing

File: functions/ieeg_getLabelXyzDestrieux.m [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+%
+
+function [labs, labs_val] = ieeg_getLabelXyzDestrieux(xyzs, FSdir, hemi, rad)
+    if nargin < 4, rad = 3; end % circle radius
+    if nargin < 3 || isempty(hemi), hemi = 'left'; end % brain hemisphere to which xyz coordinates belong 
+    
+    %% load files
+    niDestrieux = niftiRead(fullfile(FSdir, 'mri', 'aparc.a2009s+aseg.nii.gz')); % labelled nifti
+
+    [~, ~, colortable_Destrieux] = read_annotation(fullfile(FSdir, 'label', sprintf('%sh.aparc.a2009s.annot', lower(hemi(1)))));
```
Comment (dorahermes) [inline functions/ieeg_getLabelXyzDestrieux.m]: Only place here where hemi is used

**Comment by dorahermes** [comment]
I guess it currently only uses the hemi tag to load the Destrieux colortable, which is actually hemisphere agnostic. The actual code only looks for the most common label close to an electrode, so we may remove this?

**Comment by hharveygit** [comment]
Hi Dora,

Oh yes, you are correct. I'll just hardcode the left hemisphere in then.

Best,
-Harvey
________________________________

**Comment by hharveygit** [comment]
Just pushed the changes!

-Harvey
________________________________

**Comment by dorahermes** [comment]
great thanks!

**Comment by dorahermes** [comment]
thank you, this is great!

**Pull request merged**